src/python/pants/backend/awslambda/python/awslambda_python_rules.py

```diff
@@ -59,9 +59,15 @@ async def create_python_awslambda(
     python_setup: PythonSetup,
     subprocess_encoding_environment: SubprocessEncodingEnvironment,
 ) -> CreatedAWSLambda:
-    pex_filename = f"{config.address.target_name}.pex"
+    # Lambdas typically use the .zip suffix, so we use that instead of .pex.
+    pex_filename = f"{config.address.target_name}.zip"
+    # We hardcode the platform value to the appropriate one for each AWS Lambda runtime.
+    # (Running the "hello world" lambda in the example code will report the platform, and can be
+    # used to verify correctness of these platform strings.)
     py_major, py_minor = config.runtime.to_interpreter_version()
-    platform = f"linux_x86_64-cp-{py_major}{py_minor}-cp{py_major}{py_minor}m"
+    platform = f"manylinux2014_x86_64-cp-{py_major}{py_minor}-cp{py_major}{py_minor}m"
+    if (py_major, py_minor) == (2, 7):
+        platform += "u"
```

Comment:
This is tricky. Python can be either `m` or `mu`, depending on if uses 16-bit or 8-bit encoding. Are you certain that AWS Lambdas always use `mu`?

Reply:
The lambda runtime is always a very specific linux with a very specific interpreter.
These values I got from actually running the hello world lambda on every possible runtime and seeing what it printed...